rasa/core/story_conflict.py

```diff
@@ -0,0 +1,252 @@
+
+        # Iterate once more over all states and note the (unhashed) state,
+        # for which a conflict occurs
+        conflicts = StoryConflict._build_conflicts_from_states(
+            trackers, domain, max_history, rules
+        )
+
+        return conflicts
+
+    @staticmethod
+    def _find_conflicting_states(
+        trackers: List[TrackerWithCachedStates], domain: Domain, max_history: int
+    ) -> Dict[int, Optional[List[Text]]]:
+        # Create a 'state -> list of actions' dict, where the state is
+        # represented by its hash
+        rules = {}
+        for tracker, event, sliced_states in StoryConflict._sliced_states_iterator(
+            trackers, domain, max_history
+        ):
+            h = hash(str(list(sliced_states)))
+            if h not in rules:
+                rules[h] = [event.as_story_string()]
+            elif h in rules and event.as_story_string() not in rules[h]:
+                rules[h] += [event.as_story_string()]
+
+        # Keep only conflicting rules
+        return {
+            state_hash: actions
+            for (state_hash, actions) in rules.items()
+            if len(actions) > 1
+        }
+
+    @staticmethod
+    def _build_conflicts_from_states(
+        trackers: List["TrackerWithCachedStates"],
+        domain: Domain,
+        max_history: int,
+        rules: Dict[int, Optional[List[Text]]],
+    ):
+        # Iterate once more over all states and note the (unhashed) state,
+        # for which a conflict occurs
+        conflicts = {}
+        for tracker, event, sliced_states in StoryConflict._sliced_states_iterator(
+            trackers, domain, max_history
+        ):
+            h = hash(str(list(sliced_states)))
+
+            if h in rules and h not in conflicts:
+                conflicts[h] = StoryConflict(sliced_states)
+
+            if h in rules:
+                conflicts[h].add_conflicting_action(
+                    action=event.as_story_string(), story_name=tracker.sender_id
+                )
+
+        # Remove conflicts that arise from unpredictable actions
+        return [c for (h, c) in conflicts.items() if c.has_prior_events]
+
+    @staticmethod
+    def _sliced_states_iterator(
+        trackers: List[TrackerWithCachedStates], domain: Domain, max_history: int
+    ) -> (TrackerWithCachedStates, Event, List[Dict[Text, float]]):
+        """
+        Iterate over all given trackers and all sliced states within
+        each tracker, where the slicing is based on `max_history`
+        :param trackers: List of trackers
+        :param domain: Domain (used for tracker.past_states)
+        :param max_history: Assumed `max_history` value for slicing
+        :return: Yields (tracker, event, sliced_states) triplet
+        """
+        for tracker in trackers:
+            states = tracker.past_states(domain)
+            states = [
+                dict(state) for state in states
+            ]  # From rasa/core/featurizers.py:318
```

Comment:
Let's not do this comment (it's already broken)
```suggestion
            ]
```